app/controllers/password_resets_controller.rb

```diff
@@ -0,0 +1,101 @@
+class PasswordResetsController < ApplicationController
+  def new
+  end
+
+  def create
+    if params[:username].blank?
+      flash.now[:alert] = 'Please enter a username'
+      render 'new', status: :unprocessable_entity
+      return
+    end
+
+    begin
+      user = User.find(escaped_username)
+
+      if user.nil?
+        user_not_found
+      else
+        PasswordResetMailer.password_reset(user).deliver
+        flash.now[:notice] = 'Your password reset email has been sent'
+        @status = :ok
+      end
+    rescue Net::HTTPServerException => e
```

Comment:
Yeah, that's true, but moving that code outside of the `begin/rescue` block will make the resulting code more convoluted.  You'll be left with a choice of either duplicating the `render` line in multiple places, or setting a flag if there was an exception and checking for it down below, to ensure the logic around the user isn't processed if there wasn't an exception.  I see very little value in moving that code out of the `begin/rescue` block.